src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,40 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl.split('/'))?.split('.').join('.')
+
+      try {
+        const { sas: { token } } = await AzureStorage(signal).details(workspaceId)
```

Comment:
The `AzureStorage(signal).details` method makes two Ajax calls:
1. call Workspace Manager to list all the controlled resources for this workspace, then find the storage container's resourceId
2. get the SAS token for that container

Is the first ajax call absolutely necessary? Is that resource id already available/loaded somewhere in a workspace object? I don't know the answer - I'm just raising the question. It would be nice to reduce the number of ajax calls.

Reply:
I am thinking this is out of scope for this PR - but very reasonable question. @nawatts is this something that would be worth creating a ticket for to tackle later?

Reply:
The container resource ID is not available in the workspace object. It's a very fast Ajax call, so I don't think it's causing any performance issue.

Reply:
I suspect we could probably do some caching at the API level (like memoizing some API functions), but I'd be content with leaving that for a follow up. Generally, I don't think Terra UI puts much effort into minimizing API requests.

Reply:
thinking out loud here, not requesting changes for this PR: in my testing locally, it's actually the call to get the sas token that is taking 1-2 seconds and therefore causing less-than-responsive UX for the modal. And the sas token is eminently cacheable - in fact, the token even contains an expiry time.

I would love to see `const { sas: { token } } = await AzureStorage(signal).details(workspaceId)` skippable if we've already gotten a token for that workspace id, and if the token hasn't expired.

Again, I'm not requesting this change for this PR, I'm calling this out as a future optimization.

Reply:
In this particular use case (downloading a file immediately) it would be fine to cache it, but for some other uses folks want a token that they can use for some period of time. So we wouldn't want to return a token that is going to expire soon for those cases… thus I think the API would have to allow you to specify whether caching would be OK.